arkouda_viz/client/arkouda_viz/viz.py

```diff
@@ -1,97 +1,92 @@
-
-        shaded_data = ds(data_points, cmap=color_scheme).opts(
-            hv.opts.RGB(width=width, height=height)
+    if data is not None:
+        if isinstance(data, ak.DataFrame):
+            full_data = data
+
+        elif (
+            isinstance(data, tuple)
+            and len(data) == 2
+            and all(isinstance(item, ak.pdarray) for item in data)
+        ):
+            full_data = ak.DataFrame(data[0], data[1])
+
+        else:
+            raise ValueError(
+                "Invalid data. Please provide an ak.Dataframe or [ak.pdarray, ak.pdarray]."
+            )
+    else:
+        raise ValueError("Please provide data.")
+
+    cols = full_data.columns
+    initial_xrange = (ak.min(full_data[cols[0]]), ak.max(full_data[cols[0]]))
+    initial_yrange = (ak.min(full_data[cols[1]]), ak.max(full_data[cols[1]]))
+
+    def make_data(x_range, y_range, cmap):
+        if x_range is None or y_range is None or not x_range or not y_range:
+            binned_data = ak.histogram2d(
+                full_data[cols[0]], full_data[cols[1]], bins=(xBin, yBin)
+            )
+            return hv.Image(binned_data[0].to_ndarray(), bounds=(0, 0, 1, 1)).opts(
+                cmap=cmap, width=xWin, height=yWin
+            )
+        else:
+            subset_data = full_data[
+                (full_data[cols[0]] >= x_range[0])
+                & (full_data[cols[0]] <= x_range[1])
+                & (full_data[cols[1]] >= y_range[0])
+                & (full_data[cols[1]] <= y_range[1])
+            ]
+
+        x_span = x_range[1] - x_range[0]
+        y_span = y_range[1] - y_range[0]
+        binned_data = ak.histogram2d(
+            subset_data[cols[0]], subset_data[cols[1]], bins=(1000, 1000)
         )
+        return hv.Image(
+            binned_data[0].to_ndarray(),
+            bounds=(x_range[0], y_range[0], x_range[0] + x_span, y_range[0] + y_span),
+        ).opts(cmap=cmap, width=xWin, height=yWin)
 
-        shaded_data = shaded_data.opts(xlabel=f"{x_col}", ylabel=f"{y_col}")
+    class Explore(param.Parameterized):
+        cmap = param.ObjectSelector(default="viridis", objects=hv.plotting.list_cmaps())
 
-        z_score_threshold_slider.disabled = not enable_slider
+    params = Explore()
 
-        return shaded_data
-
-    logo = "../pictures/logo.png"
-    logo_image = pn.panel(logo, width=x_widget.width, height=x_widget.height)
-    layout = pn.Row(
-        pn.Column(
-            logo,
-            x_widget,
-            y_widget,
-            color_widget,
-            enable_slider_checkbox,
-            z_score_threshold_slider,
-        ),
-        update_data,
-    )
+    @pn.depends(cmap=params.param.cmap)
+    def update_plot(cmap):
+        stream = hv.streams.RangeXY(x_range=initial_xrange, y_range=initial_yrange)
+        dmap = hv.DynamicMap(
+            lambda x_range, y_range: make_data(x_range, y_range, cmap), streams=[stream]
+        )
```

Comment:
im not terribly familiar with this stuff but I imagine here or inside `make_data` is probably where you would want to do some checking to avoid recalculating the histogram when the only change is the colors. There's probably other optimizations to be made (not recalculating if someone zooms but all the points are still in the new range)

I think this can be counted as future work and I'm okay with this as a first pass